en/modules/modules_system.md

```diff
@@ -945,3 +945,50 @@ work_queue <command> [arguments...]
 
    status        print status info
 ```
+
+## internal_combustion_engine_control
+Source: [modules/internal](https://github.com/PX4/PX4-Autopilot/tree/main/src/modules/internal_combustion_engine_control)
+
+**NOTE:** This feature is not enabled by default and needs to be configured with the rpm capture driver.
+
+### Description
+The module controls the internal combustion engine ICE as described in the following architecture diagram:
+
+![Architecture](../../assets/diagrams/ice_control_diagram.png)
+
+The module publishes `InternalCombustionEngineControl.msg` containing:
+
+| Message  | Values | Explanation   |
+|------------|------------|------------|
+| `ignition_on` | True/False | Activate/deactivate |
+| `throttle_control` | [0, 1] | Motor should idle with 0. Includes slew rate if enabled. |
+| `choke_control` | [0, 1] | 1 closes the air inlet |
+| `starter_engine_control` | [0, 1] | 1 if engine start should be delayed |
+| `user_request` | On/Off | User intent |
+|   |   |  |
+
+### Implementation
+The ICE is implemented with a (4) state machine:
+
+![Architecture](../../assets/diagrams/ice_control_state_machine.png)
+
+The state machine
+- checks if `Rpm.msg` is updated to know if the engine is running
+- allows for user inputs from
+  - AUX{N}
+  - the arming state in `VehicleStatus.msg`
+  - Mavlink (TBD)
+
+
+
+<a id="internal_combustion_engine_control_usage"></a>
+### Usage
```

Comment:
FYI, you can do this like: `### Usage {#internal_combustion_engine_control_usage}`

Reply:
Actually, maybe you can't since this is auto generated.